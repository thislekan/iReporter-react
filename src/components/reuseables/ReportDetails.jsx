import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import cx from 'classnames';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import style from '../../styles/ReportDetails.css';

function ReportDetails(props) {
  const { incident, mapUrl } = props;
  return (
    <div className={style['report-body-wrapper']}>
      <div id="detailed-incident">
        <div className={cx(style['report-body-wrapper'], style.pushed)}>
          <div className={style['report-details']}>
            <div className={style['incident-intro']}>
              <FontAwesomeIcon icon='check-circle' />
              <h4>ID: </h4>
            </div>
            <p id="incident-id">{incident.id}</p>
          </div>
          <div className={style['report-details']}>
            <div className={style['incident-intro']}>
              <FontAwesomeIcon icon='check-circle' />
              <h4>Type: </h4>
            </div>
            <p id="incident-type">{incident.type}</p>
          </div>
          <div className={style['report-details']}>
            <div className={style['incident-intro']}>
              <FontAwesomeIcon icon='check-circle' />
              <h4>Created: </h4>
            </div>
            <p id="incident-date">{moment(Number(incident.createdOn)).format('MMM Do YY')}</p>
          </div>
          <div className={style['report-details']}>
            <div className={style['incident-intro']}>
              <FontAwesomeIcon icon='check-circle' />
              <h4>Status: </h4>
            </div>
            <p id="incident-status">{incident.status}</p>
          </div>
          <div className={style['report-details']}>
            <div className={style['incident-intro']}>
              <FontAwesomeIcon icon='check-circle' />
              <h4>Title: </h4>
            </div>
            <p id="incident-title">{incident.title}</p>
          </div>
          <div className={style['location-div']}>
            <div className={style['report-details']}>
              <div className={style['incident-intro']}>
                <FontAwesomeIcon icon='check-circle' />
                <h4>Location: </h4>
              </div>
              <p id="incident-location">{incident.location}</p>
            </div>
            <div className={style.map}>
              <h4>Map</h4>
              <div className={style['map-region']}>
                <iframe
                  src={mapUrl}
                  frameBorder="0"
                  width="100%"
                  height="100%"
                  title={`${incident.location}'s map`}
                  style={{ border: 0, position: 'absolute' }}
                  allowFullScreen
                />
              </div>
            </div>
            <div className={style['comment-div']}>
              <h4>Comment: </h4>
              <textarea
                name="comment"
                id="incident-comment"
                disabled
                value={(incident.comment) ? incident.comment : ''}
              />
            </div>
          </div>
          {(incident.images || incident.video) && <div className={style['image-and-video-div']}>
            <h4>Images or Videos: </h4>
            <div className={style['content-wrapper']}>
              <div id="incident-video-div"></div>
              {incident.images && incident.images.length >= 1 && (
                <div className={style['image-container']} id="incident-img-div">
                  {incident.images.map((image, index) => <img src={image} key={index} alt="" />)}
                </div>
              )}
            </div>
          </div>}
        </div>
      </div>
    </div>
  );
}

ReportDetails.propTypes = {
  incident: PropTypes.object.isRequired,
  mapUrl: PropTypes.string,
};

ReportDetails.defaultProps = {
  mapUrl: '',
};

export default ReportDetails;
