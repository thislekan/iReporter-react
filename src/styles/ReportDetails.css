.report-details{
    padding: .6rem .75rem;
    display: grid;
    grid-template-columns: 10rem auto;
    align-items: center;
    justify-content: space-between;
}
.report-details p{
    margin: .5rem 0;
    margin-left: 1.5rem;
    text-align: right;
}
.incident-intro{
    display: flex;
    align-items: center;
}
.incident-intro svg{
    color: #117d5a;
}
.map-region{
    height: 20rem;
    border: 1px solid;
    position: relative;
}
.map{
    margin: 2rem 0;
}
.content-wrapper{
    height: 100%;
    margin: 1rem 0;
}
.image-container{
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-template-rows: auto;
    grid-gap: 1rem;
    margin-top: 1rem;
}
.image-container img{
    height: 14.5rem;
    width: 100%;
    object-fit: scale-down;
    object-position: center;
    border: 1px solid rgba(29, 57, 74, 0.35);
    border-radius: .25rem;
}
@media only screen and (min-width: 700px) {
  .map-region{
        height: 30rem;
    }
    .image-container img{
        height: 12rem;
    }
}
@media only screen and (min-width: 900px){
  .image-container img{
        height: 22rem;
    }
}