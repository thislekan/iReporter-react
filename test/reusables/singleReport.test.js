import React from 'react';
import { shallow } from 'enzyme';
import ReportDetails from '../../src/components/reuseables/SingleReport.jsx';
import incident from '../mock/incident';

const mock = incident;
const mockFn = jest.fn();

describe('<SingleReport />', () => {
  it('should render SingleReport', () => {
    const wrapper = shallow(
      <ReportDetails
        location={{ pathname: '/user' }}
        id={mock.id}
        incidentLocation={mock.location}
        type={mock.type}
        createdOn={mock.createdOn}
        alterDeleteModal={mockFn}
        alterEditModal={mockFn}
        status={mock.status}
        fetchIncident={mockFn}
        title={mock.title}
      />,
    );
    expect(wrapper).toMatchSnapshot();
    wrapper.find('#view_btn').simulate('click', { fetchIncident: f => f });
    wrapper.find('#edit_btn').simulate('click', { alterEditModal: f => f });
    wrapper.find('#delete_btn').simulate('click', { alterDeleteModal: f => f });
    expect(mockFn).toHaveBeenCalledTimes(3);
  });
  it('should render a different content when route is Admin', () => {
    const wrapper = shallow(
      <ReportDetails
        location={{ pathname: '/admin' }}
        id={mock.id}
        incidentLocation={mock.location}
        type={mock.type}
        createdOn={mock.createdOn}
        alterDeleteModal={mockFn}
        alterEditModal={mockFn}
        status={mock.status}
        fetchIncident={mockFn}
        title={mock.title}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
