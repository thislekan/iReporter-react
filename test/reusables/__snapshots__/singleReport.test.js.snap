// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleReport /> should render SingleReport 1`] = `
<div>
  <div>
    <div>
      <div>
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="check-circle"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
        <h4>
          Title: 
        </h4>
      </div>
      <p>
        dhjfd
      </p>
    </div>
    <div>
      <div>
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="check-circle"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
        <h4>
          Type: 
        </h4>
      </div>
      <p>
        red-flag
      </p>
    </div>
    <div>
      <div>
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="check-circle"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
        <h4>
          Date: 
        </h4>
      </div>
      <p
        id="incident-date"
      >
        Invalid date
      </p>
    </div>
    <div>
      <div>
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="check-circle"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
        <h4>
          Status: 
        </h4>
      </div>
      <p>
        draft
      </p>
    </div>
    <div>
      <div>
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="check-circle"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
        <h4>
          Location: 
        </h4>
      </div>
      <p>
        sdjnkjs
      </p>
    </div>
    <div>
      <div>
        <button
          id="view_btn"
          onClick={[Function]}
        >
          Details
        </button>
        <button
          id="edit_btn"
          onClick={[Function]}
        >
          Edit
        </button>
        <button
          id="delete_btn"
          onClick={[Function]}
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<SingleReport /> should render a different content when route is Admin 1`] = `
<div>
  <div>
    <div>
      <div>
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="check-circle"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
        <h4>
          Title: 
        </h4>
      </div>
      <p>
        dhjfd
      </p>
    </div>
    <div>
      <div>
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="check-circle"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
        <h4>
          Type: 
        </h4>
      </div>
      <p>
        red-flag
      </p>
    </div>
    <div>
      <div>
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="check-circle"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
        <h4>
          Date: 
        </h4>
      </div>
      <p
        id="incident-date"
      >
        Invalid date
      </p>
    </div>
    <div>
      <div>
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="check-circle"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
        <h4>
          Status: 
        </h4>
      </div>
      <p>
        draft
      </p>
    </div>
    <div>
      <div>
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="check-circle"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
        <h4>
          Location: 
        </h4>
      </div>
      <p>
        sdjnkjs
      </p>
    </div>
    <div>
      <div>
        <button
          id="view_btn"
          onClick={[Function]}
        >
          Details
        </button>
        <button
          id="edit_btn"
          onClick={[Function]}
        >
          Edit
        </button>
      </div>
    </div>
  </div>
</div>
`;
